docs/developer-docs/security/security-best-practices/inter-canister-calls.mdx

```diff
@@ -269,11 +279,71 @@ mod test {
 }
 ```
 
+</TabItem>
+
+<TabItem value="motoko" label="Motoko">
+
+```motoko
+import Result "mo:base/Result";
+import Map "mo:base/HashMap";
+import Error "mo:base/Error";
+import Principal "mo:base/Principal";
+
+actor {
+
+  let pending_requests = Map.HashMap<Principal, Bool>(10, Principal.equal, Principal.hash);
+
+  private func guard(principal : Principal) : Result.Result<Principal, Error> {
```

Comment:
Do we need the `Principal` as result or should we switch to `()`? IIUC the return value is simply ignored further down.

Reply:
I return the `Result<Principal, Error>` value in case people want to use this as a template and extend it. But if you think it's not necessary I can remove it.

Reply:
I'd suggest to use Result<(), Error> because the result is currently unused and for me it was slightly confusing as it doesn't seem needed.